pkg/operator/ceph/cluster/osd/spec.go

```diff
@@ -156,8 +157,10 @@ func (c *Cluster) makeDeployment(nodeName string, selection rookalpha.Selection,
 
 	// Set osd memory target to the best appropriate value
 	if !osd.IsFileStore {
-		if !c.resources.Limits.Memory().IsZero() {
-			commonArgs = append(commonArgs, "--osd-memory-target", strconv.Itoa(int(c.resources.Limits.Memory().Value())))
+		// As of Nautilus Ceph auto-tunes its osd_memory_target on the fly so we don't need to force it
```

Comment:
Does Ceph auto tune its osd_memory_target by looking at the cgroups limits? If not, I think we still want this.

Reply:
Yes, it tunes `osd_memory_target` based on the cgroup limit.

Reply:
was this behavior added in Nautilus?  I was using Mimic (because Nautilus wouldn't start for me).  Will cc Mark Nelson.

Reply:
https://github.com/ceph/ceph/pull/26856

Reply:
Hi Folks,

Only comment I have is to please make sure there is some amount of extra memory available.  We can't guarantee that the RSS memory usage will always be under the osd_memory_target for a variety of reasons.  The big one is that there is no guarantee that the kernel will reclaim memory that the OSD process has unmapped.  Kernel settings such as transparent_huge_pages can have an effect here as well.  Typically I don't see the OSD consuming more than 10% extra, but 20% might provide a little extra breathing room.  It's possible we can do some things to reduce this further in the future though we'll never be able to make any guarantees.

Mark

Reply:
Thanks @markhpc, so I believe the 80% ratio is good.